src/main/scala/com/gu/subscriptions/cas/service/SubscriptionService.scala

```diff
@@ -0,0 +1,95 @@
+package com.gu.subscriptions.cas.service
+
+import com.gu.membership.zuora.ZuoraApiConfig
+import com.gu.membership.zuora.soap.Zuora._
+import com.gu.membership.zuora.soap._
+import com.gu.monitoring.ZuoraMetrics
+import com.gu.subscriptions.cas.config.Configuration
+import com.gu.subscriptions.cas.model.SubscriptionExpiration
+import com.gu.subscriptions.cas.service.utils.ScheduledTask
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.concurrent.duration._
+
+class SubscriptionService(zuoraClient: ZuoraClient, knownProducts: List[String]) {
+  import spray.json._
+  import DefaultJsonProtocol._
+
+  case class SubsRequest(subscriberId: Option[String], password:String)
+  implicit val subsRequestFormat = jsonFormat2(SubsRequest)
+
+  def extractZuoraSubscriptionId(subsRequest: SubsRequest): Option[String] =
+    subsRequest.subscriberId.filter(_.startsWith("A-S"))
+
+  def extractSubsRequest(requestJson: String): SubsRequest = {
+    requestJson.parseJson.convertTo[SubsRequest]
+  }
+
+  val samePostcode: (String, String) => Boolean = {
+    val format: String => String = _.replaceAll("\\s+", "").toLowerCase
+
+    (postcodeA, postcodeB) => format(postcodeA) == format(postcodeB)
+  }
+
+  def verifySubscriptionExpiration(subscriptionName: String, postcode: String): Future[SubscriptionExpiration] =
+    for {
+      subscription <- zuoraClient.queryForSubscription(subscriptionName)
+
+      ratePlan <- zuoraClient.queryForRatePlan(subscription.id)
+      productRatePlan <- zuoraClient.queryForProductRatePlan(ratePlan.productRatePlanId)
+      product <- zuoraClient.queryForProduct(productRatePlan.productId)
+        if knownProducts.contains(product.name)
```

Comment:
New products can also be added, but adding new products or changing the name of existing ones will be dealt in the same way. I am inclined to favour this approach because it's agnostic to which Zuora environment the service connects to and it makes it slightly easier to add new products (in a single place). Also it's easier to reason about names than about ids.